R/private.R

```diff
@@ -0,0 +1,187 @@
+# @file general
+#
+# Copyright 2020 Observational Health Data Sciences and Informatics
+#
+# This file is part of R OHDSI webApi package.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#private function recursively converts variables that are NULL to NA
+.convertNullToNARecursive <- function(x) {
+  nullToNARecursive <- function(x, fn)
+  {
+    if(is.list(x)) {
+      lapply(x, nullToNARecursive, fn)
+    } else if (is.data.frame(x)) {
+      x[is.null(x)] <- NA
+    } else {
+      fn(x)
+    }
+  }
+  nullToNARecursive(x, function(x) if(is.null(x)) NA else x)
+}
+
+
+#compares a reference vector of objects to a reference vector. If different it will return the difference.
+.isVectorContained = function(vectorToCompare = c(), vectorReference = c()){
+  x = vector(length = length(vectorToCompare))
+  for (i in 1:length(vectorToCompare)) {
+      x[i] = vectorToCompare[i] %in% vectorReference
+      if (length(which(vectorToCompare[i] %in% vectorReference)) == 0) {
+        vectorReference } else {
+          vectorReference = vectorReference[-match(vectorToCompare[i], vectorReference)]
+        }
+  }
+  if (all(x == T)) {
+    TRUE
+  } else {
+    return(vectorToCompare[!x])
+  }
+}
+
+# recursively flattens tree based structure.
+.flattenTree <- function(node, accumulated) {
+  if (is.null(node$children)) {
+    accumulated$name <- c(accumulated$name, node$name);
+    accumulated$size <- c(accumulated$size, node$size);
+    return(accumulated)
+  } else {
+    for (child in node$children) {
+      accumulated <- flattenTree(child, accumulated)
+    }
+    return(accumulated)
+  }
+}
+
+# converts time in integer/milliseconds to date-time with timezone.
+# assumption is that the system timezone = time zone of the local server running webApi.
+.millisecondsToDate <- function(milliseconds) {
+  sec <- milliseconds/1000
+  as.POSIXct(sec, origin = "1970-01-01", tz = Sys.timezone())
+}
+
+# checks if url conforms with expected structure for base url
+.checkWebApiBaseUrl <- function(baseUrl){
```

Comment:
Why the new function? Was the existing one insufficient? If so, let's modify the existing function (.checkBaseUrl).

Reply:
sure - lets deprecate the original function and replace with this one. I will rename the new one to .checkBaseUrl. I will comment out the old one - we can delete it later.